objects/ctrl_game/Draw_0.gml

```diff
@@ -0,0 +1,16 @@
+/// @description Draw pause screen
+if (!global.gamePaused) {
```

Comment:
Stahp with all the global variables :-)

Draw events and Steps are called every step, so you're running through this if statement 60 times per second, even when the game is not paused.

It might be nicer to have an overlay for the Pause screen already drawn but set to visible=false, then set it to visible=true in objects/ctrl_game/KeyPress_80.gml (and back to false when continuing the game).